pkg/webhooks/jobset_webhook.go

```diff
@@ -162,6 +163,21 @@ func (j *jobSetWebhook) ValidateCreate(ctx context.Context, obj runtime.Object)
 	}
 
 	var allErrs []error
+
+	// Validate InPlaceRestart feature gate.
+	// The in-place restart API should be used only when the feature gate is enabled.
+	if !features.Enabled(features.InPlaceRestart) {
+		if js.Spec.FailurePolicy != nil && js.Spec.FailurePolicy.RestartStrategy == jobset.InPlaceRestart {
+			allErrs = append(allErrs, fmt.Errorf("InPlaceRestart restart strategy cannot be set when InPlaceRestart feature gate is disabled"))
```

Comment:
I am wondering what if users disable validation webhooks, Kubernetes doesn't provide any additional functionality to detect if users try to use feature that is under FeatureGate?

Reply:
>I suspect this marker is used for informing 

Thanks for letting me know!

>What do you think about adding a log or event if a JobSet with in-place restart is reconciled while the feature gate is disabled?

I think it is fair to say that it is user responsibility to understand what features are available if they disable webhook.
@kannon92 @mimowo @tenzen-y Do you log any messages in Kueue when users try to use feature without feature gate enabled?

Reply:
I don't think so. A log isn't a bad idea to have.

I think in k8s we may just silently ignore that field. We get in scenarios like this sometimes if apiserver and/or kubelet both need the feature gate and one is disabled. So the apiserver could have the field but kubelet won't obey it.